test/inf-clojure-tests.el

```diff
@@ -0,0 +1,86 @@
+;;; inf-clojure-tests.el --- Tests for Inf-Clojure -*- lexical-binding: t; -*-
+;;
+;; Copyright © 2014-2018 Bozhidar Batsov
+
+;; Authors: Bozhidar Batsov <bozhidar@batsov.com>
+;;          Andrea Richiardi <a.richiardi.work@gmail.com>
+;; URL: http://github.com/clojure-emacs/inf-clojure
+;; Keywords: processes, clojure
+;; Package-Requires: ((emacs "24.4") (clojure-mode "5.3"))
+
+;; This file is part of GNU Emacs.
+
+;; GNU Emacs is free software: you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; GNU Emacs is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with GNU Emacs.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+;;
+;; Code completion using alexander-yakushev/compliment.
+
+;;; Code:
+
+;; Tests for inf-clojure.el
+
+(message "Running tests on Emacs %s" emacs-version)
+
+(require 'buttercup)
+(require 'assess)
+(require 'inf-clojure)
+
+(cl-defmacro ict-with-assess-buffers ((&rest varlist) &body body)
+  `(assess-with-temp-buffers (,@varlist)
+                             (clojure-mode)
+                             (inf-clojure-minor-mode)
+                             ,@body))
+
+(defun ict-bounds-string (bounds)
+  (buffer-substring (car bounds) (cdr bounds)))
+
+(describe "completion bounds at point" ()
+
+          (it "computes bounds for plain-text"
+              (ict-with-assess-buffers
+               ((a (insert "plain-text")))
+               (with-current-buffer a
+                 (expect (ict-bounds-string (inf-clojure-completion-bounds-of-expr-at-point))
+                         :to-equal "plain-text"))))
+
+          (it "computes bounds for @deref when point is after it"
+              (ict-with-assess-buffers
+               ((a (progn
+                     (insert "@deref")
+                     (goto-char 7))))
+               (with-current-buffer a
+                 (expect (ict-bounds-string (inf-clojure-completion-bounds-of-expr-at-point))
+                         :to-equal "deref"))))
+
+          (it "computes bounds for ^:keyword when point is after it"
```

Comment:
This is failing still, but I moved to `buttercup` and `assess`

Reply:
What is it actually matching?

Reply:
it should compute `:keyword` without the hat

Reply:
Now it is returning `nil`

Reply:
And what happens when you try this interactively?

Reply:
I tried not in a test, in normal buffer it was working

Reply:
So if you open an `inf-clojure-minor-mode` buffer and put this in it:

```
^:pr|
```

In `ielm` you'll see:

```
ELISP> (inf-clojure-completion-bounds-of-expr-at-point)
(2 . 5)
```